be/src/storage/rowset/rowset_writer.cpp

```diff
@@ -573,15 +573,12 @@ Status HorizontalRowsetWriter::_final_merge() {
         ChunkIteratorPtr itr;
         // create temporary segment files at first, then merge them and create final segment files if schema change with sorting
         if (_context.schema_change_sorting) {
-            if (_context.tablet_schema->keys_type() == KeysType::DUP_KEYS) {
+            if (_context.tablet_schema->keys_type() == KeysType::DUP_KEYS ||
```

Comment:
You have to describe the problem and solution clearly in the commit message.

Reply:
added.